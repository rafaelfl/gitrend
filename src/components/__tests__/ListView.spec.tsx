import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ListView } from '..';

describe('ListView component tests', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    test('it should match snapshot', () => {
        const { container } = render(
            <ListView>
                <button>Button1</button>
                <button>Button2</button>
                <button>Button3</button>
            </ListView>,
        );

        expect(container).toMatchSnapshot();
    });

    test('it should render a list of three buttons', () => {
        render(
            <ListView>
                <button>Button1</button>
                <button>Button2</button>
                <button>Button3</button>
            </ListView>,
        );

        expect(screen.getAllByRole('button')).toHaveLength(3);
    });

    test('it should call the onClick event of the last button', () => {
        const onClick = jest.fn();

        render(
            <ListView>
                <button onClick={onClick}>Button1</button>
                <button onClick={onClick}>Button2</button>
                <button onClick={onClick}>Button3</button>
            </ListView>,
        );

        const buttons = screen.getAllByRole('button');

        expect(buttons).toHaveLength(3);

        userEvent.click(buttons[2]);
        expect(onClick).toBeCalledTimes(1);
    });

    test('it should call the scrollBy event when the ListView is scrolled', () => {
        const onWheel = jest.fn();
        window.HTMLElement.prototype.scrollBy = onWheel;

        const { getByTestId, unmount } = render(
            <ListView>
                <button>Button1</button>
                <button>Button2</button>
                <button>Button3</button>
            </ListView>,
        );

        const scrollSlots = getByTestId('@ListView/scrollSlot');

        expect(scrollSlots).toBeInTheDocument();

        fireEvent.wheel(scrollSlots, { deltaY: +100 });
        fireEvent.wheel(scrollSlots, { deltaY: -100 });

        expect(onWheel).toBeCalled();

        unmount();
    });
});
