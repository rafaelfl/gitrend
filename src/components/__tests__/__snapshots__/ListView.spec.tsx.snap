// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView component tests it should match snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL dvUyWp"
  >
    <div
      class="sc-gsDKAQ jMWrOX"
    >
      <button>
        Button1
      </button>
      <button>
        Button2
      </button>
      <button>
        Button3
      </button>
    </div>
    <button
      class="sc-hKwDye iMvXya"
    >
      <i
        class="fas fa-angle-left"
      />
    </button>
    <button
      class="sc-dkPtRN fyxqWQ"
    >
      <i
        class="fas fa-angle-right"
      />
    </button>
  </div>
</div>
`;
